versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -632,12 +641,16 @@ protected Stream<Difference> buildDiff(
               if (f.equals(t)) {
                 return null;
               }
+              byte payload =
+                  fromVal != null
+                      ? fromVal.getPayload()
+                      : (toVal != null ? toVal.getPayload() : (byte) 0);
```

Comment:
Should we check that payloads match if both are set?

Reply:
Actually a different problem.
I'll add a TODO there

Reply:
#4949